### DISCUSSION TITLE
Cache media (mp3) via cloudflare

### DISCUSSION THREAD

**Discussion by scouech** [directus/directus#6691] [Archive]
Hi!

I've set up a cache everything rule (see below) on cloudflare but it doesn't seem to cache anything... Would be awesome if you could share what you have done to cache media files (such as mp3) on cloudflare!

I've tried:

`/assets/*`
`/assets/*?download`

Cheers
![Screenshot 2021-07-07 at 10 30 12 AM](https://user-images.githubusercontent.com/26603052/124727006-98bd2180-df0e-11eb-9a30-6345cffb7be1.png)

**Comment by rijkvanzanten** [comment]
Are your files publicly readable? Directus will add the `private` flag to the cache control header when you're using an authenticated user

**Reply by scouech** [reply]
mmm files aren't private now. Good point! Is there a way to limit access to the files while allowing caching?

**Reply by rijkvanzanten** [reply]
> Is there a way to limit access to the files while allowing caching?

You could add a custom field to directus_files that acts as a "publicly accessible" flag, and then use that in the permissions rules for the public role 🤔

**Reply by scouech** [reply]
I think you lost me here :D Any chance you could give a bit more details? hehe

**Reply by rijkvanzanten** [reply]
You can set the permissions for files for the public role based on a [filter rule](https://docs.directus.io/reference/filter-rules/#filter-rules) structure, so one way to allow the world to access a file (and thus have Cloudflare cache it) without having public access to every file is by adding a custom field to directus_files (like `publicly_accessible`), and then using a permissions rule like

```json
{
  "publicly_accessible": {
    "_eq": true
  }
}
```

That way, you can choose what files can be downloaded/cached on a case-by-case basis 🙂

**Reply by scouech** [reply]
Thanks @rijkvanzanten.

So just to be sure, I should process in this order?
1. Add a new field for `directus_files` such as `publicly_accessible` with `true` as a default value
2. Give rights to the public role to read files
3. Add a filter rule to public role file reading to
```
{
  "publicly_accessible": {
    "_eq": true
  }
}
```

But that would still give access to the public to all the files with the custom field with the `true `value right?

**Reply by rijkvanzanten** [reply]
Yeah! That's one way to do it (assuming you don't want every file to be publicly accessible of course, if you don't have any private files, you could simply enable full read access to the public role)

**Reply by scouech** [reply]
mmm i see! Makes sense! And is there a way to no give access to /files/  (where you get the entire files list) but only to the individual files for the public role? 
It would be hard for the public role to access a specific file with its ID, however it would be really easy if the role can access the entire files list..

**Reply by rijkvanzanten** [reply]
Not at this time. That being said, it has come up a couple times before, so it's not unthinkable a dedicated "browse vs open" permission will be added soon (cc @benhaynes), ref https://github.com/directus/directus/discussions/6365

**Reply by benhaynes** [reply]
Yeah, splitting read permissions (or extending support) to have both **List** and **Read** (individual) is definitely something we've had requested a lot, and we want to implement. 👍

**Reply by scouech** [reply]
ok cool thanks both @rijkvanzanten @benhaynes. 

In the meantime, it means I have no proper way to do what I'm looking to do right? Could I somehow redirect people if they try to reach /files but not if they try to reach /files/ID ? A bit of a messy solution I guess

**Reply by rijkvanzanten** [reply]
Hmm not that I can think of. You might be able to hack in a redirect using a hook, but I'm not sure!
